core/citrus-api/src/main/java/com/consol/citrus/validation/AbstractMessageValidator.java

```diff
@@ -39,11 +39,14 @@
     @Override
     public final void validateMessage(Message receivedMessage, Message controlMessage, TestContext context,
             List<ValidationContext> validationContexts) throws ValidationException {
-        T validationContext = findValidationContext(validationContexts);
-
-        // check if we were able to find a proper validation context
-        if (validationContext != null) {
-            validateMessage(receivedMessage, controlMessage, context, validationContext);
+        
+        // validate all matching contexts
+        for (ValidationContext validationContext : validationContexts) {
+            if (getRequiredValidationContextType().isInstance(validationContext)) {
+                if (validationContext != null) {
```

Comment:
actually it cannot be `null` because `Object#isInstance` would not match otherwise. the second `if`-clause is therefore useless. from the docs:
> The method returns true if the specified Object argument is **non-null** and can be cast [...]